oddoor/models/oddoor_lock.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class OddoorLock(models.Model):
+
+    _name = 'oddoor.lock'
+    _description = 'Oddoor Lock'
+
+    name = fields.Char()
+    description = fields.Text()
+    group_ids = fields.Many2many(
+        'oddoor.group',
+    )
+    action_ids = fields.One2many(
+        'oddoor.key.action',
+        inverse_name='lock_id',
+    )
+    active = fields.Boolean(default=True, required=True)
+
+    @api.multi
+    def check_access_unique_virtual_key(self, unique_virtual_key):
+        self.ensure_one()
+        key = self.env['oddoor.key'].search([
+            ('unique_virtual_key', '=', unique_virtual_key),
+        ], limit=1)
+        if not key:
+            return False
```

Comment:
Why not also creating a refused action if someone tries to enter with a non existing key?

Reply:
True, fixed